### PR TITLE
Update Alpha menu - refactor, bugfix, and QOL updates

### LINKED ISSUE #1297: Intermittent - Selector moves above # on #-Z menu in movies library
## Software Versions

- Jellyfin Server Version:  0.8.10 - on local network
- Roku Client Version:  Version 1.6 - build 6 - on local network

## Describe the bug
Intermittent - Selector moves above # on #-Z menu in movies library

## How To Reproduce
1. Go to 'My Media -> {Movies Library}'
2. Click on '#-Z' at seemingly random 
3. Scroll around through the movies DO NOT press select on a movie stay in the movies library menu
4. Repeat 2-3 at random
6. Press left to the #-Z menu and then up to the top of the menu
7. Bug occurs - Selector moves above # and becomes an empty square, when selected it returns the Z filtered results

## Expected behavior
Selector should not move above #

## Message
Again, sorry I can't reliably reproduce this, unlike #1295 which I've seen more than a few times, I've only seen this one once.

### LINKED ISSUE #1525: Alpha menu mic icon can be focused and selected
## Software Versions

- Jellyfin Server Version:  10.8.12
- Roku Client Version:  unstable as of 11/21/23

## Describe the bug
Alpha menu on itemgrid allows you to highlight and select the mic icon (sometimes not all the time). when selected, it filters the media based on the letter Z

## How To Reproduce
I don't know the exact path. It seemed spratic but eventually I can reproduce by just scrolling all the way back up the alpha text

1. Open a movie library
2. Go left to focus the alpha text
3. Select a letter
4. Go right to focus the itemgrid
5. Go left to focus the alpha text
6. Scroll all the way up
7. Icon might be focusable? If not keep doing these steps or something similar until it is

## Expected behavior
mic icon is never focusable. it is just there to tell the user they can use their remote's mic button to select a letter(which works great btw)

## Screenshots
Notice the area above the # symbol looks off. that is because the mic icon is focused (and has been selected)
alpha-menu-mic-icon-bug

## Connection Information

- Is server local or remote?
- Is server connection HTTP or HTTPS?

### LINKED ISSUE #1526: Using mic for Alpha menu doesn't move focus to the letter being filtered
## Is your feature request related to a problem? Please describe
When using the mic with the Alpha Menu shown on the Itemgrid, after speaking your selection the itemgrid updates as expected but the current focus in the alpha text never updates to reflect the letter spoken.

## Describe the solution you'd like
Change focus to the letter spoken by the user

### LINKED ISSUE #1527: Selecting the same letter multiple times from the alpha menu does nothing
## Is your feature request related to a problem? Please describe
When using the alpha menu shown on the itemgrid, if I have my library filtered by say letter A for example, when I select the letter A again from the alpha menu nothing happens

## Describe the solution you'd like
If the letter selected is already being used as a filter, then selecting it again should toggle the filter back off.

## Additional context
The magic "rewind 15 seconds" button has the same functionality but I don't find that intuitive and frequently forget that's an option

### LINKED ISSUE #1528: Keep focus on letter selected when using alpha menu
## Is your feature request related to a problem? Please describe
After selecting a letter from the alpha menu shown on the itemgrid, the focus switches to the list of items in the itemgrid no matter how many items there are. This is worst when the filtered list of media contains no items as the focus gets "lost" on the itemgrid and the user must go back left to the alpha menu (or hit the magic rewind 15 sec button) to regain focus. 

I also find this a little annoying when the filtered list of media contains more than 0 items because it takes me a second to realize what has happened. The letter I selected is still highlighted but is grey instead of white so at first I think my focus is there before realizing I have the first item selected.

## Describe the solution you'd like
Keep focus on the letter. That way if no results are returned, I can continue scrolling up/down and select a different letter. And if results are returned, keep focus on the letter to simplify the UX.

Also, don't let the focus leave the alpha menu when there are 0 results returned.

### LINKED ISSUE #1767: Deselect Alpha Picker
## Is your feature request related to a problem? Please describe
Currently the alpha picker can be used to filter by a character, such as `A-Z`. However unlike web version, clicking the same letter a second time does not clear the selection.

## Describe the solution you'd like
It would be nice if, clicking the selected character in the alpha picker, would deselect the selection and allow for scrolling through the unfiltered library again

### PR DESCRIPTION
**Pull request by cewert**
This converts the alpha menu to use the correct styling for vars, it moves the mic icon inside the alpha node, fixes an edge case bug, and adds some quality of life improvements.

## Changes
- Refactor `Alpha.bs` and all files that use it to use lowerCamelCase for all var names
- Move the mic icon from `ItemGrid`, `MovieLibraryView`, and `MusicLibraryView` and put it inside the Alpha component. ~~Also removes the mic text completely~~
  - Moved the mic icon and mic text to the overhang to match the tvguide page (except for music library which hides the overhang)
  - The tvguide/itemgrid was the only thing using the mic text (and reusing the mic icon). I am now creating the mic icon and text nodes as needed for the livetv guide view.
- Convert mic icon from a `Button` to a `Poster` node
  - It was difficult to reproduce in the first place but I haven't been able to reproduce #1525 since making this change
- Keep focus on alpha menu after selecting a letter
  - EDIT: I'm keeping focus on alpha menu now whenever it is active no matter what. Including after using the voice filter option
- When a user selects the same letter that is already focused, remove the current filter instead of doing nothing
- Move logic to show mic icon into `Alpha.bs`
- When using the voice search/filter feature, if the users search is for a single character, select the corresponding option on the alphamenu i.e. saying "Z" will select "Z" on the alhpamenu. saying 3 will select "#" etc.

## Issues
Fixes #1297
Fixes #1526 
Fixes #1527 
Fixes #1528

Note: there was one thing I tried to fix but couldn't figure out how:
- When the alpha menu is not focused, there is a wide "focused" overlay that shows up on top of it although it is hard to see with the naked eye. Here is a screenshot after first load of the movie library (notice the overlay around #)
image

### CODE REVIEW

**Comment by 1hitsong** [comment]
In my Movie library, in Presentation view, I clicked a letter and it threw an error on line 737 `m.spinner.visible = true`

image

**Comment by cewert** [comment]
hah so the reason I missed that crash was because of my movie library being bugged (it would load using the itemgrid instead of the movie specific grid). Apparently that fixed itself recently because I can reproduce this crash. Great timing 😆

**Comment by 1hitsong** [comment]
This also likely fixes #1297

File: components/ItemGrid/MusicLibraryView.bs [hunk 1]
```diff
@@ -323,14 +325,14 @@ end sub
 '
 'Handle loaded data, and add to Grid
 sub ItemDataLoaded(msg)
-    stopLoadingSpinner()
-    m.top.alphaActive = false
     itemData = msg.GetData()
     m.loadItemsTask.unobserveField("content")
     m.loadItemsTask.content = []
 
+    print "itemData=", itemData
```
Comment (neilsb) [inline components/ItemGrid/MusicLibraryView.bs]: This just a missed debugging statement, or wanting to leave it in?

Comment (cewert) [reply components/ItemGrid/MusicLibraryView.bs]: Missed it. Thanks

**Review by jimdogx — APPROVED** [review]

**Pull request merged**